File: keyboardwatcher/src/main/java/com/kirchhoff/keyboardwatcher/KeyboardWatcherActivity.java
package com.kirchhoff.keyboardwatcher;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.azimolabs.keyboardwatcher.KeyboardWatcher;

public class KeyboardWatcherActivity extends AppCompatActivity implements KeyboardWatcher.OnKeyboardToggleListener {

    private KeyboardWatcher keyboardWatcher;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.a_keyboard_watcher);

        keyboardWatcher = new KeyboardWatcher(this);
        keyboardWatcher.setListener(this);
    }

    @Override
    public void onKeyboardShown(int keyboardSize) {
        Log.d("TAG", "onKeyboardShown, size = " + keyboardSize);
    }

    @Override
    public void onKeyboardClosed() {
        Log.d("TAG", "onKeyboardClosed");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        keyboardWatcher.destroy();
    }
}

File: keyboardwatcher/src/main/java/com/kirchhoff/keyboardwatcher/KeyboardWatcherActivity.kt
package com.kirchhoff.keyboardwatcher

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log

import com.azimolabs.keyboardwatcher.KeyboardWatcher

class KeyboardWatcherActivity : AppCompatActivity(), KeyboardWatcher.OnKeyboardToggleListener {

    private lateinit var keyboardWatcher: KeyboardWatcher

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.a_keyboard_watcher)

        keyboardWatcher = KeyboardWatcher(this)
        keyboardWatcher.setListener(this)
    }

    override fun onKeyboardShown(keyboardSize: Int) {
        Log.d("TAG", "onKeyboardShown, size = $keyboardSize")
    }

    override fun onKeyboardClosed() {
        Log.d("TAG", "onKeyboardClosed")
    }

    override fun onDestroy() {
        super.onDestroy()
        keyboardWatcher.destroy()
    }
}
